src/App.tsx

```diff
@@ -16,80 +19,171 @@ export const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPHABET,
-  LENGTH,
+  NONE = 'none',
+  ALPHABET = 'alpabet',
+  LENGTH = 'length',
 }
 
-type ReorderOptions = {
+type State = {
   sortType: SortType,
   isReversed: boolean,
 };
 
 // Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
-  { sortType, isReversed }: ReorderOptions,
+  { sortType, isReversed }: State,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((productName1, productName2) => (
+        productName1.localeCompare(productName2)
+      ));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((product1, product2) => (
+        product1.length - product2.length
+      ));
+      break;
+
+    default:
+      break;
```

Comment:
I think we can add that NONE type will be default)

case SortType.NONE:
default:
break;